### ISSUE TITLE
Alpine/pmOS OS-release app not adopted

### ISSUE THREAD

**Issue by pabloyoyoista** [Cogitri/gnome-software-plugin-apk#38]
If we ever want to do OS upgrades, the OS-release app created by `core/gs-plugin-os-release` is specially useful. Right now, we adopt it, but ignore it and treat like other regular apps:

```
16:38:29:0791 GsPluginApk Adding updates
16:38:29:0793 GsPluginApk Starting refinining process
16:38:29:0793 GsPluginApk App */*/*/system/* has quirk WILDCARD or is of REPOSITORY kind; skipping!
16:38:29:0793 GsPluginApk Ignoring app */*/*/org.alpinelinux.alpine-3.15.0/*, not owned by apk
16:38:29:0793 Gs  returning imperfect match: */*/*/system/* != */*/*/org.alpinelinux.alpine-3.15.0/*
16:38:29:0793 Gs  Chaining cancellation from 0x7f7a4d3b42a0 to 0x7f7a4cbde7a0
16:38:29:0793 GsPluginApk Adopted app */*/*/org.alpinelinux.alpine-3.15.0/*
16:38:29:0793 Gs  apk adopted */*/*/org.alpinelinux.alpine-3.15.0/*
16:38:29:0795 GsPluginApk Starting refinining process
16:38:29:0795 GsPluginApk app */*/*/org.alpinelinux.alpine-3.15.0/* has missing sources; skipping
16:38:29:0795 Gs  app invalid as no summary */*/*/org.alpinelinux.alpine-3.15.0/*
16:38:29:0795 Gs  running refine with dedupe-flags=7 with refine-flags=require-size,require-version,require-update-details,require-icon,require-runtime with interactive=True with timeout=60, elapsed time since creation 1ms
```

**Comment by DylanVanAssche** [comment]
Do you know what this app does? I have been trying to wrap my head around this but couldn't...

**Comment by pabloyoyoista** [comment]
No, I am really not sure. I just figured out that there is one specific plugin whose goal is to create it, so (I assumed) it must have some role in the identification process of the distro and release being run. Could you maybe ask upstream?

**Comment by DylanVanAssche** [comment]
According to Adrian Vovk on upstream's Matrix channel:

> It looks like it sets up a package by consuming info in /etc/os-release. This means gnome software will treat the OS itself as an app. Not sure what exactly it's useful for, but presumably it'll make an app called tld.domain.osname-version (so org.fedora.fedora-36). I guess if you point gnome software at that app ID, it'll show a details page containing the information parsed out of /etc/os-release
> I suppose you could try and use it to identify the distro or something like that. Or adopt the app and then mark _it_ as updatable in g-s. I've been getting away without using it in my updater

So we don't need it actually... It looks at the distro as yet-another-app which may be updatable. However, most distros use the big banner to announce big updates, so I would rather go with that.

**Comment by pabloyoyoista** [comment]
Ok, thanks, that's good to know, one less thing to care about! Digging a bit into the code and the history of the plugin, I found that it is used to provide info on [historical updates](https://gitlab.gnome.org/GNOME/gnome-software/-/blob/main/plugins/packagekit/gs-plugin-packagekit.c#L2927-L2940), so that after the upgrade, the user is somewhat notified about it. This is totally cosmetic and maybe not applicable to our case if we do "online" updates, but could be a nice cosmetic still worth to have an issue about :)

**Comment by DylanVanAssche** [comment]
Agreed :) For now, we don't prioritize this for OS upgrades (yet).
Not sure if we can make it, but having OS upgrades working for postmarketOS v22.06 (Alpine 3.16) would be awesome

**Comment by pabloyoyoista** [comment]
>  Not sure if we can make it, but having OS upgrades working for postmarketOS v22.06 (Alpine 3.16) would be awesome

I would say that is too risky, and an unnecessary push. We might be able to write the code, but I'm very hesitant that we would be able to fix all the bugs. If you the CLI updates for v22.06 is successful and battle-tested by users, then next release cycle it should be much safer to go ahead with graphical updates

**Comment by pabloyoyoista** [comment]
Example of something that could go wrong: https://gitlab.alpinelinux.org/Cogitri/apk-polkit-rs/-/issues/6

**Comment by DylanVanAssche** [comment]
Yeah, you're right... Forget what I said :P 
Oooof that's a big issue :(
